lib/src/working_copy.rs

```diff
@@ -602,13 +602,14 @@ impl TreeState {
         let sparse_matcher = self.sparse_matcher();
         let current_tree = self.store.get_tree(&RepoPath::root(), &self.tree_id)?;
         let mut tree_builder = self.store.tree_builder(self.tree_id.clone());
-        let mut deleted_files: HashSet<_> = self
-            .file_states
-            .iter()
-            .filter_map(|(path, state)| {
-                (state.file_type != FileType::GitSubmodule).then(|| path.clone())
-            })
-            .collect();
+        let mut deleted_files: HashSet<_> = trace_span!("deleting file states").in_scope(|| {
```

Comment:
This step just iterates over an in-memory map and creates a set from it, so is it worth instrumenting it? If we do want to keep it, I think we should at least change the description, because "deleting file states" makes it sound like it does more work than it does.

Reply:
I could rename it `filtering git submodules`?

Reply:
Or "collecting existing files"?